src/io.rs

```diff
@@ -2,85 +2,126 @@
 
 use core::{fmt, slice};
 use core::fmt::Write;
+use core::ptr;
 
 /// File descriptors
-const STDOUT: usize = 1;
-const STDERR: usize = 2;
+static mut STDOUT: isize = -1;
+static mut STDERR: isize = -1;
 
 /// Host's standard error
 struct Stderr;
 
 /// Host's standard output
 struct Stdout;
 
-fn write_all(fd: usize, mut buffer: &[u8]) {
+pub fn get_stdout() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDOUT)}
+}
+
+pub fn get_stderr() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDERR)}
+}
+
+/// Open stdout and stderr.
+pub fn open_streams() -> Result<(),()> {
```

Comment:
>  I assume it's OK to open e.g. stdout more than once

No, it's not OK, I've seen debuggers that does not like closing and reopening stdout.